### DISCUSSION TITLE
How to Implement IMemoryCache in a Server That Hosts Multiple Applications?

### DISCUSSION THREAD

**Discussion by sohlae** [dotnet/AspNetCore.Docs#23977] [Q&A]
According to the [documentation](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-6.0), implementing IMemoryCache is something like this.

```C#
_memoryCache.GetOrCreate("cache_key",
    cacheEntry =>
    {
        cacheEntry.SlidingExpiration = TimeSpan.FromSeconds(3);
        return DateTime.Now;
    });
```

My question is, if I am deploying my application in a server that hosts several other applications, and say another application decided to store something in the cache using the key "cache_key", will it overwrite my cache data? If yes, is it a good practice then to give a more specific name for the cache key?

**Comment by Rick-Anderson** [comment]
Not if it's another app.
